Fix build on Plasma 6.3

AldanTanneo opened the pull request:
Updates the blur effect to be compatible with Plasma 6.3 by copying upstream changes.

Should fix #140, #143, #145, #147, #154, #155

AldanTanneo committed:
Fix #140: apply same changes as upstream kwin blur effect

scgm0 commented:
This fixes the compilation bugs, but the performance degradation is noticeable and seems to be an issue with kde 6.1, the upstream version had performance issues in 6.1 as well...

akano12 commented:
> This fixes the compilation bugs, but the performance degradation is noticeable and seems to be an issue with kde 6.1, the upstream version had performance issues in 6.1 as well...

Skill issue

akano12 commented:
> Should also fix #143 

Thanks man!

RomanKode commented:
Can confirm, the changes work, everything should be good.

Plasma 6.1

scgm0 commented:
> Skill issue

`Skill` issue?
Sorry my English is not good enough to understand what it means.

AldanTanneo committed:
keep blur effect up to date with upstream kwin effect

AldanTanneo commented:
Latest push fixed breakage with Plasma 6.2, I don't advise updating if you're still on 6.1

AldanTanneo committed:
remove uneeded enabled by default logic

Threkork commented:
Thank you so mach🙏

SakuraPuare committed:
Update KDecoration references from KDecoration2 to KDecoration3

- Update .gitignore to include additional build and IDE directories
- Modify CMakeLists.txt to find KDecoration3 instead of KDecoration2
- Update blur effect source code to use KDecoration3 headers and namespaces
- Add XCB::XCB as a private link library in blur CMakeLists.txt

akano12 commented:
Bro can you fix it for plasma 6.3 if possible? Thanks in advance

AldanTanneo committed:
nitpick: align to upstream plasma effect

AldanTanneo commented:
Updated to Plasma 6.3 and KDecoration3 (I took a fix from a fork of my fork, funny how it goes)

scgm0 commented:
> Updated to Plasma 6.3 and KDecoration3 (I took a fix from a fork of my fork, funny how it goes)

Thanks!
